Description: Ignore SIGPIPE from choosers
 which means they closed all fds to the pipe, i.e. they don't need more data
 (e.g. `head -n1` used in tests/choose.rs). Just carry on.
Forwarded: https://github.com/casey/just/issues/2635
Last-Update: 2025-02-21
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/subcommand.rs
+++ b/src/subcommand.rs
@@ -242,3 +242,3 @@
     for recipe in recipes {
-      writeln!(
+      if let Err(e) = writeln!(
         child.stdin.as_mut().unwrap(),
@@ -246,7 +246,10 @@
         recipe.namepath.spaced()
-      )
-      .map_err(|io_error| Error::ChooserWrite {
-        io_error,
-        chooser: chooser.clone(),
-      })?;
+      ) {
+        if e.kind() != std::io::ErrorKind::BrokenPipe {
+          return Err(Error::ChooserWrite {
+            io_error: e,
+            chooser: chooser.clone(),
+          });
+        }
+      }
     }
